tests/scalers/aws/aws_cloudwatch_ignore_null_values_false/aws_cloudwatch_ignore_null_values_false_test.go

```
@@ -0,0 +1,196 @@
+        dimensionValue: {{.CloudWatchMetricDimensionValue}}
+        metricName: {{.CloudWatchMetricName}}
+        targetMetricValue: "1"
+        minMetricValue: "1"
+        ignoreNullValues: "false"
+        metricCollectionTime: "120"
+        metricStatPeriod: "60"
+`
+)
+
+var (
+	testNamespace                  = fmt.Sprintf("%s-ns", testName)
+	deploymentName                 = fmt.Sprintf("%s-deployment", testName)
+	scaledObjectName               = fmt.Sprintf("%s-so", testName)
+	secretName                     = fmt.Sprintf("%s-secret", testName)
+	cloudwatchMetricName           = fmt.Sprintf("cw-%d", GetRandomNumber())
+	awsAccessKeyID                 = os.Getenv("TF_AWS_ACCESS_KEY")
+	awsSecretAccessKey             = os.Getenv("TF_AWS_SECRET_KEY")
+	awsRegion                      = os.Getenv("TF_AWS_REGION")
+	cloudwatchMetricNamespace      = "DoesNotExist"
+	cloudwatchMetricDimensionName  = "dimensionName"
+	cloudwatchMetricDimensionValue = "dimensionValue"
+	minReplicaCount                = 0
+)
+
+// This test is to verify that the scaler results in an error state when
+// the metric query returns null values and the ignoreNullValues is set to false.
+func TestCloudWatchScalerWithIgnoreNullValuesFalse(t *testing.T) {
+	ctx := context.Background()
+
+	// setup cloudwatch
+	cloudwatchClient, err := cloudwatch.NewClient(ctx, awsRegion, awsAccessKeyID, awsSecretAccessKey, "")
+	assert.Nil(t, err, "error creating cloudwatch client")
+
+	// check that the metric in question is not already present, and is returning
+	// an empty set of values.
+	metricQuery := cloudwatch.CreateMetricDataInputForEmptyMetricValues(cloudwatchMetricNamespace, cloudwatchMetricName, cloudwatchMetricDimensionName, cloudwatchMetricDimensionValue)
+	metricData, err := cloudwatch.GetMetricData(ctx, cloudwatchClient, metricQuery)
+	assert.Nil(t, err, "error getting metric data")
+	assert.Nil(t, cloudwatch.ExpectEmptyMetricDataResults(metricData), "metric data should be empty")
```

```suggestion
	require.Nil(t, err, "error getting metric data")
	require.Nil(t, cloudwatch.ExpectEmptyMetricDataResults(metricData), "metric data should be empty")
```
We try to fail fast when the test requirements have failed